src/interfaces/permissions-grant.ts

```diff
@@ -104,4 +106,34 @@ export class PermissionsGrant extends Message<PermissionsGrantMessage> {
       );
     }
   }
+
+  public static validateScope(permissionsGrantMessage: PermissionsGrantMessage): void {
+    const scope = permissionsGrantMessage.descriptor.scope;
+    if (scope.interface === DwnInterfaceName.Records) {
+      // RecordsRead and RecordsWrite
+      if (scope.method === DwnMethodName.Read || scope.method === DwnMethodName.Write) {
```

Comment:
Sanity: this two outer most `ifs` seem to read: "scope can be anything if interface is not `Records`" and "scope can be anything if methods are not read or write".

I removed the two outer `ifs` and all tests pass.

Consider just remove since condition is not reachable, or throw error in `else` if intent is defensive programming.

Reply:
I'll remove for now. But here's some context: Grants for other DWN methods have scopes that can only contain `interface` and `method`. This is validated in JSON schema. In other words, grant scopes for DWN methods other than `RecordsRead` and `RecordsWrite` currently shouldn't ever have any of the fields checked inside the outer two `ifs`.

Reply:
Removed